hotsos/core/plugins/kernel/net.py

```diff
@@ -303,24 +303,27 @@ def __init__(self):
         if "UDP" not in self._data:
             self._data["UDP"] = {}
 
-        # Read other relevant values from sysctl
-        (udp_mem_min, udp_mem_pressure, udp_mem_max) = getattr(
-            SYSCtlFactory(), 'net.ipv4.udp_mem').split(" ")
-        (tcp_mem_min, tcp_mem_pressure, tcp_mem_max) = getattr(
-            SYSCtlFactory(), 'net.ipv4.tcp_mem').split(" ")
-
-        self._data["TCP"]["sysctl_mem_min"] = int(tcp_mem_min)
-        self._data["UDP"]["sysctl_mem_min"] = int(udp_mem_min)
-        self._data["TCP"]["sysctl_mem_pressure"] = int(tcp_mem_pressure)
-        self._data["UDP"]["sysctl_mem_pressure"] = int(udp_mem_pressure)
-        self._data["TCP"]["sysctl_mem_max"] = int(tcp_mem_max)
-        self._data["UDP"]["sysctl_mem_max"] = int(udp_mem_max)
-        self._data["TCP"]["statistics_mem_usage_pct"] = round(
-            (self.GlobTcpSocksTotalMemPages / self.SysctlTcpMemMax
-             if self.SysctlTcpMemMax else 0) * 100.0, 2)
-        self._data["UDP"]["statistics_mem_usage_pct"] = round(
-            (self.GlobUdpSocksTotalMemPages / self.SysctlUdpMemMax
-             if self.SysctlUdpMemMax else 0) * 100.0, 2)
+        def maybe_parse_xmem(key):
```

Comment:
can we please avoid using nested functions?

Reply:
I like scoping related things together, but of course we can :+1: